### PR TITLE
ARROW-3368: [Integration/CI/Python] Add dask integration test to docker-compose setup

### PR DESCRIPTION
**Pull request by kszucs**
Port dask integration testing, and refactored the hdfs one. 
Multiple python hdfs tests cases are failing, nut sure why.

### CODE REVIEW

File: python/testing/dask_tests/test_dask_integration.py [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-            the_date = start_date
-            for _ in range(rowcount):
-                temperature = randint(-10, 35)
-                dewpoint = temperature - randint(0, 10)
-                writer.writerow({'date': the_date, 'temperature': temperature,
-                                 'dewpoint': dewpoint})
-                the_date += timedelta(days=1)
-
-
-def test_dask_file_read(tmpdir):
```
Comment (kszucs) [inline python/testing/dask_tests/test_dask_integration.py]: AFAICS it doesn't test interoperability between dask and pyarrow, rather `dask <-> pandas` and `pandas <-> pyarrow`

File: integration/dask/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (wesm) [inline integration/dask/Dockerfile]: Oof, I missed that we were putting integration testing unrelated to the Arrow binary protocol in integration/. I don't think it makes sense to organize the tests like this. 

https://issues.apache.org/jira/browse/ARROW-3975

Might make sense to put the protocol tests under `format/`

File: integration/hdfs/Dockerfile [hunk 3]
```diff
@@ -15,63 +15,34 @@
...
 
+# installing libhdfs (JNI)
+ARG HADOOP_VERSION=2.6.0
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 \
+    HADOOP_HOME=/usr/local/hadoop \
+    HADOOP_OPTS=-Djava.library.path=/usr/local/hadoop/lib/native \
+    PATH=$PATH:$HADOOP_HOME/bin:$HADOOP_HOME/sbin
+RUN apt-get update -y && \
+    apt-get install -y openjdk-8-jdk && \
+    wget http://archive.apache.org/dist/hadoop/core/hadoop-$HADOOP_VERSION/hadoop-$HADOOP_VERSION.tar.gz && \
```
Comment (wesm) [inline integration/hdfs/Dockerfile]: We aren't supposed to be using the dist system like this. Is there another way to download the libraries that either uses the Apache mirror system, or to get them from a package manager?

**Review by wesm — COMMENTED** [review]
A few comments; I am not sure anything has to be done this second about ARROW-3975 but let me know if there is another option for procuring the HDFS client dependencies

File: integration/dask/Dockerfile (hunk 2 above)
Comment (kszucs) [reply integration/dask/Dockerfile]: I thought integration is the right place for 3rd party integration tests as well (I mean it's still better then `dev`).

File: integration/hdfs/Dockerfile (hunk 3 above)
Comment (kszucs) [reply integration/hdfs/Dockerfile]: Right, let me take a look.

Comment (kszucs) [reply integration/hdfs/Dockerfile]: Updated

**Comment by kszucs** [comment]
Running tests: https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-371
Python HDFS tests will abort.

**Comment by wesm** [comment]
What's wrong with the Python tests?

**Comment by wesm** [comment]
Needs rebase

**Comment by kszucs** [comment]
@wesm rebased, but the python tests are still failing. Resolve them in a follow-up PR to not block this PR?

**Comment by wesm** [comment]
Sure no problem. Can you open a follow up JIRA?

**Comment by kszucs** [comment]
Builds: [kszucs/crossbow/build-381](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-381)

**Comment by kszucs** [comment]
JIRA ticket: [ARROW-4053](https://issues.apache.org/jira/browse/ARROW-4053)
Crossbow builds: [build-384](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-384)

**Review by wesm — APPROVED** [review]
+1. Thanks @kszucs! I haven't resolved the Docker problems on my machine yet but I'll take this for a spin when I can

**Pull request closed without merging**